db/column_family.cc

```diff
@@ -1566,14 +1566,15 @@ Status ColumnFamilyData::SetOptions(
 }
 
 // REQUIRES: DB mutex held
-Env::WriteLifeTimeHint ColumnFamilyData::CalculateSSTWriteHint(int level) {
+Env::WriteLifeTimeHint ColumnFamilyData::CalculateSSTWriteHint(Version* version,
+                                                               int level) {
   if (initial_cf_options_.compaction_style != kCompactionStyleLevel) {
     return Env::WLTH_NOT_SET;
   }
   if (level == 0) {
     return Env::WLTH_MEDIUM;
   }
-  int base_level = current_->storage_info()->base_level();
+  int base_level = version->storage_info()->base_level();
```

Comment:
Given that this function's only reliance on CFD is for the immutable options, it seems like this CalculateSSTWriteHint function should be elsewhere. It looks like VersionStorageInfo has some related functions that take ImmutableOptions. Something to consider.

Reply:
Thank you for this idea! Looks like it's also only relying on the compaction style in the immutable options, which VersionStorageInfo also keeps track of. So I have moved this function to be a member of `VersionStorageInfo`.